test/api_container_attach_test.go

```diff
@@ -18,8 +21,24 @@ func (suite *APIContainerAttachSuite) SetUpTest(c *check.C) {
 	SkipIfFalse(c, environment.IsLinux)
 }
 
-// TestContainerAttachOk tests attaching containers is OK.
-func (suite *APIContainerAttachSuite) TestContainerAttachOk(c *check.C) {
+// TestContainerAttachStdin tests attaching stdin is OK.
+func (suite *APIContainerAttachSuite) TestContainerAttachStdin(c *check.C) {
 	// TODO
 	// path := "/containers/{name:.*}/attach"
 }
+
+// TestContainerAttachNotFound
+func (suite *APIContainerAttachSuite) TestContainerAttachNotFound(c *check.C) {
+	cname := "TestContainerAttachNotFound"
+
+	q := url.Values{}
+	q.Set("stdin", "1")
+	query := request.WithQuery(q)
+	header := request.WithHeader("Content-Type", "text/plain")
+
+	resp, err := request.Post("/containers/"+cname+"/attach", query, header)
+	c.Assert(err, check.NotNil)
+
+	// TODO: return 404 when issue 470 is fixed.
+	CheckRespStatus(c, resp, 200)
```

Comment:
I think this is reasonable. Since if someone fix the bug, and this test case will no longer pass. Then he needs to update the test case.